### PR TITLE
Add systemd support in deb

### PR DESCRIPTION
**Pull request by sleirsgoevy**

### CODE REVIEW

File: deb/scripts/postinst [hunk 1]
```diff
@@ -1,8 +1,9 @@
 #!/bin/bash
```
Comment (MikailBag) [inline deb/scripts/postinst]: Should it really be non-executable?

File: deb/build.sh [hunk 2]
```diff
@@ -75,16 +75,31 @@ tar -xvf "$JJS_TGZ"
 mv jjs pkg
 
 mkdir data
```
Comment (MikailBag) [inline deb/build.sh]: Now useless?

File: deb/build.sh [hunk 3]
```diff
@@ -75,16 +75,31 @@ tar -xvf "$JJS_TGZ"
 mv jjs pkg
```
Comment (MikailBag) [inline deb/build.sh]: Why?

File: deb/jjs-oneclick [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+echo '# Setting up database'
+if [ -f /var/lib/jjs/db-setup.sql ]
+then # upgrade
+    set -e
+    dd if=/usr/share/jjs/db-setup.sql bs=$(stat -c %s /var/lib/jjs/db-setup.sql) skip=1 | su postgres -c 'psql -U postgres -d jjs -a -f /dev/stdin'
+    set +e
+else # first install
+    set -e
+    su postgres -c 'psql -U postgres -c "create role jjs with password '"'"'internal'"'"';"'
```
Comment (MikailBag) [inline deb/jjs-oneclick]: If CREATE USER, LOGIN is enabled automatically.

File: deb/jjs-oneclick [hunk 5]
```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+
```
Comment (MikailBag) [inline deb/jjs-oneclick]: Set -e ?

File: deb/env.txt [hunk 6]
```diff
@@ -0,0 +1,6 @@
+JJS_SYSROOT=/var/lib/jjs
+JJS_PATH=/usr
```
Comment (MikailBag) [inline deb/env.txt]: /opt/jjs

File: deb/env.txt [hunk 7]
```diff
@@ -0,0 +1,6 @@
...
+DATABASE_URL=postgres://jjs:internal@localhost:5432/jjs
```
Comment (MikailBag) [inline deb/env.txt]: Add TODO: we should default to unix socket instead

File: deb/env.txt [hunk 8]
```diff
@@ -0,0 +1,6 @@
...
+JJS_HOST=0.0.0.0
+JJS_SELF_ADDR=127.0.0.1
+JJS_ENV=dev
```
Comment (MikailBag) [inline deb/env.txt]: production

File: deb/jjs-oneclick [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+if [ "$(whoami)" != root ]
+then
+    echo 'jjs-oneclick must be run as root!'
+    exit 1
+fi
+which psql >/dev/null || exit $?
+
+echo '# Setting up config'
```
Comment (MikailBag) [inline deb/jjs-oneclick]: Use jjs-setup instead

File: deb/jjs-oneclick [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+su -c 'jjs-setup --data-dir /var/lib/jjs --install-dir /usr/share/jjs --setup-config' -s "$SHELL" jjs
+
+echo '# Setting up problems'
```
Comment (MikailBag) [inline deb/jjs-oneclick]: Use jjs-setup instead

File: deb/jjs-oneclick [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+    for i in *
+    do
+        echo "* $i"
+        mkdir /var/lib/jjs/var/problems/"$i"
+        JJS_PATH=/var/lib/jjs jjs-ppc compile --pkg "$i" --out /var/lib/jjs/var/problems/"$i"
+    done
+    ' -s "$SHELL" jjs
+)
+
+echo '# Setting up database'
```
Comment (MikailBag) [inline deb/jjs-oneclick]: Use jjs-setup instead

File: deb/jjs-oneclick (hunk 10 above)
Comment (MikailBag) [reply deb/jjs-oneclick]: Well, for this case you can add TODO instead because setup is unable to partially update DB

File: deb/build.sh (hunk 2 above)
Comment (sleirsgoevy) [reply deb/build.sh]: No

File: deb/build.sh (hunk 3 above)
Comment (sleirsgoevy) [reply deb/build.sh]: It's a package, isn't it?

File: deb/jjs-oneclick (hunk 5 above)
Comment (sleirsgoevy) [reply deb/jjs-oneclick]: I think it shouldn't be set, so e.g. database setup will still run if sysroot setup fails.

File: deb/env.txt (hunk 6 above)
Comment (sleirsgoevy) [reply deb/env.txt]: What's wrong with installing into /usr?

Comment (MikailBag) [reply deb/env.txt]: Because it is wasting OS with rubbish like `/usr/etc/example-config`

File: deb/jjs-oneclick (hunk 5 above)
Comment (MikailBag) [reply deb/jjs-oneclick]: Is there situation when creating  sysroot failed for legitimate reason, so we should continue setup?

File: deb/build.sh (hunk 2 above)
Comment (MikailBag) [reply deb/build.sh]:
```suggestion
mkdir -p data/opt
mv jjs data/opt/jjs
mkdir -p data/lib/systtemd/system
```

File: deb/jjs-oneclick (hunk 11 above)
Comment (sleirsgoevy) [reply deb/jjs-oneclick]: Not possible until it supports partial db setup

File: deb/env.txt (hunk 7 above)
Comment (sleirsgoevy) [reply deb/env.txt]: Is it valid to have comments in systemd env. files?

File: deb/build.sh (hunk 2 above)
Comment (sleirsgoevy) [reply deb/build.sh]: I think this is semantically incorrect, since data will also contain other directories

Comment (MikailBag) [reply deb/build.sh]: For example?

File: deb/env.txt (hunk 7 above)
Comment (MikailBag) [reply deb/env.txt]: Well, add to any other place then:)

File: deb/build.sh (hunk 2 above)
Comment (sleirsgoevy) [reply deb/build.sh]: All system directories with symlinks to /opt/jjs

File: deb/env.txt (hunk 7 above)
Comment (sleirsgoevy) [reply deb/env.txt]: Anyway I don't understand why unix socket is better than loopback TCP. Postgres still does have proper authorization.

**Pull request merged**